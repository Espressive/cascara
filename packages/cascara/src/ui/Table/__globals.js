import pt from 'prop-types';
import { dataModules } from '../../modules/ModuleKeys';

const dataModuleOptions = Object.keys(dataModules);

export const PROP_TYPES = {
  // An array of action objects that will show up on the right side of a table row
  actions: pt.arrayOf(pt.shape({})),

  // An array of data objects that can potentially show up in the table.
  data: pt.arrayOf(pt.shape({})),

  // An ordered array of modules configured to show a corresponding `data` object
  dataDisplay: pt.arrayOf(
    pt.shape({
      module: pt.oneOf(dataModuleOptions).isRequired,
    })
  ),

  // The object property in a `data` object used to identify its uniqueness
  uniqueIdAttribute: pt.string,
};

export const UUID_PRIORITY_KEYS = [
  'eid',
  'uuid',
  'id',
  'sys_date_created',
  'number',
];

export const INFER_UNIQUE_ID = (objectKeys) => {
  for (const key of UUID_PRIORITY_KEYS) {
    if (objectKeys.includes(key)) {
      return key;
    }
  }
  return undefined;
};
