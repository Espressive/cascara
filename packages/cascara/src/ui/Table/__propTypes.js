import pt from 'prop-types';
import { actionModules, dataModules } from '../../modules/ModuleKeys';

const actionModuleOptions = Object.keys(actionModules);
const dataModuleOptions = Object.keys(dataModules);

export const TABLE_SHAPE = {
  /** Actions will be appended to each row, they'll appear as buttons. */
  actions: pt.shape({
    actionButtonMenuIndex: pt.number,

    modules: pt.arrayOf(
      pt.shape({
        module: pt.oneOf(actionModuleOptions).isRequired,
      })
    ),

    // Resolve record actions.
    // A function that returns the actions available to the current row
    resolveRecordActions: pt.func,
  }),

  // An array of objects.
  //
  // Every object in this array will potencially be rendered as a table row.
  data: pt.arrayOf(pt.shape({})),

  // DEPRECATED: The main configuration for your table. Here you can specify the columns to display
  // as well as the available actions (if any) for each row.
  dataConfig: pt.shape({
    /** DEPRECATED - use actions instead */
    actionButtonMenuIndex: pt.number,

    /** DEPRECATED - use actions instead */
    actions: pt.arrayOf(
      pt.shape({
        module: pt.oneOf(actionModuleOptions).isRequired,
      })
    ),

    /** DEPRECATED dataDisplay instead */
    display: pt.arrayOf(
      pt.shape({
        module: pt.oneOf(dataModuleOptions).isRequired,
      })
    ),
  }),

  /** Here you can describe each of the visible columns in your table. */
  dataDisplay: pt.arrayOf(
    pt.shape({
      module: pt.oneOf(dataModuleOptions).isRequired,
    })
  ),

  // Event handler.
  //
  // An event handler you can pass to handle every event your table emits.
  onAction: pt.func,

  // Resolve record actions.
  // A function that returns the actions available to the current row
  resolveRecordActions: pt.func,

  // Unique ID Attribute.
  //
  // specifies the attribute that uniquely identifies every object in the 'data' array.
  uniqueIdAttribute: pt.string,
};
