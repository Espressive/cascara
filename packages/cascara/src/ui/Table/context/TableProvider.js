import React from 'react';
import { ModuleContext, ModuleProvider } from '../../../modules/context';
import { useForm } from 'react-hook-form';

const TableProvider = ({ children, value, ...props }) => {
  const formMethods = useForm();
  const { handleSubmit } = formMethods;

  const onSubmit = (data) => {
    // eslint-disable-next-line no-console
    console.table(data);

    /** @brian:
     *
     * I am thinking on a way for us to derive the type of action
     * to rise here, depending on which mode we are in:
     *
     * isEditing -> update
     * isCreating -> create
     */
    value.onSubmit(data);
  };

  const mergedValues = {
    ...ModuleContext.defaultValue,
    ...value,
    formMethods,
  };

  return (
    <ModuleProvider value={mergedValues} {...props}>
      <form onSubmit={handleSubmit(onSubmit)}>{children}</form>
    </ModuleProvider>
  );
};

export default TableProvider;
