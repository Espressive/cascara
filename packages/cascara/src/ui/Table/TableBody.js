import React, { useContext } from 'react';
import { ModuleContext } from '../../modules/context';
import styles from './Table.module.scss';

import { Boundaries } from '../../system-components';
import TableRow from './TableRow';

const TableBody = () => {
  const { data, dataDisplay, uniqueIdAttribute } = useContext(ModuleContext);

  const rows = data?.map((rowData) => ({
    columns: dataDisplay?.map((itemConfig) => ({
      ...itemConfig,
    })),
    id: data[uniqueIdAttribute],
    rowData,
  }));

  return (
    <Boundaries>
      <tbody className={styles.BodyContainer}>
        {rows?.map(({ rowData }, i) => (
          <TableRow key={i} rowData={rowData} />
        ))}
      </tbody>
    </Boundaries>
  );
};

export default TableBody;
