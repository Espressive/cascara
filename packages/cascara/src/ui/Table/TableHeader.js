import React, { useContext } from 'react';

import { ModuleContext } from '../../modules/context';

const TableHeader = () => {
  const { dataConfig } = useContext(ModuleContext);
  const headerCells = dataConfig.display.map((column) => (
    <th key={column.attribute}>{column.label}</th>
  ));

  if (dataConfig.actions && dataConfig.actions.length) {
    headerCells.push(<th key={'action-bar-slot'} />);
  }

  return (
    <thead>
      <tr>{headerCells}</tr>
    </thead>
  );
};

export default TableHeader;
