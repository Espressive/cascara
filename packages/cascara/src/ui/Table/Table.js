import React from 'react';
import pt from 'prop-types';
import styles from './Table.module.scss';

import ErrorBoundary from '../../shared/ErrorBoundary';
import TableProvider from './context/TableProvider';

import TableHeader from './TableHeader';
import TableBody from './TableBody';

import { actionModules, dataModules } from '../../modules/ModuleKeys';

const actionModuleOptions = Object.keys(actionModules);
const dataModuleOptions = Object.keys(dataModules);

const propTypes = {
  /** An array of objects.
   *
   * Every object in this array will potencially be rendered as a table row. */
  data: pt.arrayOf(pt.shape({})),

  /** The main configuration for your table. Here you can specify the columns to display
   * as well as the available actions (if any) for each row. */
  dataConfig: pt.shape({
    /** Actions will be appended to each row, they'll appear as buttons. */
    actions: pt.arrayOf(
      pt.shape({
        module: pt.oneOf(actionModuleOptions).isRequired,
      })
    ),

    /** Here you can describe each of the visible columns in your table. */
    display: pt.arrayOf(
      pt.shape({
        module: pt.oneOf(dataModuleOptions).isRequired,
      })
    ),
  }),

  /** Event handler.
   *
   * An event handler you can pass to handle every event your table emits.*/
  onAction: pt.func,

  /** Unique ID Attribute.
   *
   * specifies the attribute that uniquely identifies every object in the 'data' array. */
  uniqueIdAttribute: pt.string,
};

/** This is a Table */
const Table = ({
  data = [],
  dataConfig = {},
  onAction = (type, data) => type,
  uniqueIdAttribute,
  ...rest
}) => {
  const { actions = [], display = [] } = dataConfig;
  let columnCount = display.length;

  if (actions.length) {
    columnCount++;
  }

  return (
    <ErrorBoundary>
      <TableProvider
        value={{
          data,
          dataConfig,
          onAction,
          uniqueIdAttribute,
        }}
        {...rest}
      >
        <table
          className={styles.Table}
          style={{
            gridTemplateColumns: `repeat(${columnCount}, auto)`,
          }}
        >
          <TableHeader />
          <TableBody />
        </table>
      </TableProvider>
    </ErrorBoundary>
  );
};

Table.propTypes = propTypes;

export default Table;
