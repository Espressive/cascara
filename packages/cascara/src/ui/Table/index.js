import React from 'react';

import { TableContextProvider } from './context/TableContext';
import Table from './Table';

export default ({ data, dataConfig, onAction }) => (
  <TableContextProvider data={data} dataConfig={dataConfig} onAction={onAction}>
    <Table />
  </TableContextProvider>
);
