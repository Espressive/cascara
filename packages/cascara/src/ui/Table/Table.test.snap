// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table component tree snapshot test 1`] = `
<div>
  <form>
    <table
      class="Table"
      style="grid-template-columns: repeat(5, auto);"
    >
      <thead
        class="HeadContainer"
      >
        <tr
          class="Row"
        >
          <th
            class="HeadCell"
          >
            Active
          </th>
          <th
            class="HeadCell"
          >
            ID
          </th>
          <th
            class="HeadCell"
          >
            Email
          </th>
          <th
            class="HeadCell"
          >
            Country
          </th>
          <th
            class="HeadCell"
          />
        </tr>
      </thead>
      <tbody
        class="BodyContainer"
      >
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                  data-checked="true"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  024f2316-265a-46e8-965a-837e308ae678
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Hayden.Zieme@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  Argentina
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  8f676552-3b91-4474-a9d8-a27892bc4d87
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Hubert_Denesik@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  Argentina
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                  data-checked="true"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  4189e6e0-2033-4017-b027-7a345a79e0c8
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Kathryn.Bernier40@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  Argentina
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                  data-checked="true"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  fdb9b932-f30b-44e6-ac03-9d5f9269c096
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Vito4@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  USA
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                  data-checked="true"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  35ef7196-c587-44f3-81ab-d732ce143614
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Damaris_Bednar52@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  Argentina
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;

exports[`Table component tree without row actions 1`] = `
<div>
  <form>
    <table
      class="Table"
      style="grid-template-columns: repeat(5, auto);"
    >
      <thead
        class="HeadContainer"
      >
        <tr
          class="Row"
        >
          <th
            class="HeadCell"
          >
            Active
          </th>
          <th
            class="HeadCell"
          >
            ID
          </th>
          <th
            class="HeadCell"
          >
            Email
          </th>
          <th
            class="HeadCell"
          >
            Country
          </th>
          <th
            class="HeadCell"
          />
        </tr>
      </thead>
      <tbody
        class="BodyContainer"
      >
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                  data-checked="true"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  024f2316-265a-46e8-965a-837e308ae678
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Hayden.Zieme@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  Argentina
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  8f676552-3b91-4474-a9d8-a27892bc4d87
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Hubert_Denesik@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  Argentina
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                  data-checked="true"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  4189e6e0-2033-4017-b027-7a345a79e0c8
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Kathryn.Bernier40@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  Argentina
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                  data-checked="true"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  fdb9b932-f30b-44e6-ac03-9d5f9269c096
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Vito4@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  USA
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
        <tr
          class="Row"
        >
          <td
            class="Cell"
          >
            <div
              class="Checkbox"
            >
              <span>
                <span
                  class="Input"
                  data-checked="true"
                />
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Text"
            >
              <span>
                <span
                  class="Input"
                >
                  35ef7196-c587-44f3-81ab-d732ce143614
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Email"
            >
              <span>
                <span
                  class="Input"
                >
                  Damaris_Bednar52@espressive.com
                </span>
              </span>
            </div>
          </td>
          <td
            class="Cell"
          >
            <div
              class="Select"
            >
              <span>
                <span
                  class="Input"
                >
                  Argentina
                </span>
              </span>
            </div>
          </td>
          <td
            class="CellActions"
          >
            <button
              aria-label="Edit"
              class="ui basic icon button"
              name="edit.start"
              type="button"
            >
              <svg
                focusable="false"
                height="1em"
                preserveAspectRatio="xMidYMid meet"
                style="transform: rotate(360deg); vertical-align: -0.143em;"
                viewBox="0 0 140 140"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M87.89 23.252a2.917 2.917 0 0 0-2.064-.864a2.917 2.917 0 0 0-2.065.858L20.049 86.975a2.917 2.917 0 0 0 0 4.124l28.863 28.863a2.917 2.917 0 0 0 4.125 0l63.682-63.682a2.917 2.917 0 0 0 0-4.118zM14.175 98a2.917 2.917 0 0 0-2.852-.74a2.917 2.917 0 0 0-2.048 2.123L.49 135.998a2.917 2.917 0 0 0 .776 2.742a2.958 2.958 0 0 0 2.742.77l36.586-8.75a2.917 2.917 0 0 0 1.383-4.9zm121.158-80.943L122.95 4.667a14.583 14.583 0 0 0-20.603 0l-8.272 8.265a2.917 2.917 0 0 0 0 4.125l28.875 28.869a2.917 2.917 0 0 0 4.124 0l8.26-8.278a14.583 14.583 0 0 0 0-20.591z"
                  fill="currentColor"
                />
              </svg>
            </button>
            <button
              class="ui basic button"
              name="view"
              type="button"
            >
              view
            </button>
            <button
              class="ui basic button"
              name="delete"
              type="button"
            >
              delete
            </button>
          </td>
        </tr>
      </tbody>
    </table>
  </form>
</div>
`;
